transport/socks5/socks5.go

```diff
@@ -11,6 +11,15 @@ import (
 	"strconv"
 )
 
+// AuthMethod is the authentication method as defined in RFC 1928 section 3.
+type AuthMethod = uint8
+
+// SOCKS authentication methods as defined in RFC 1928 section 3.
+const (
+	AuthNo               AuthMethod = 0x00
```

Comment:
I think we can rename these methods like this:
```go
const (
	MethodNoAuth uint8 = iota
	MethodGSSAPI
	MethodUserPass
	MethodNoAcceptable = 0xFF
)
```

Reply:
I renamed and added GSSAPI and MethodNoAcceptable, the rest I don't see how a chance how they are ever going to be implemented.